src/utils.ts

```diff
@@ -13,9 +13,9 @@ import {type WallpaperManager} from './manager/wallpaperManager.js';
 // 'SourceType' is already declared in the upper scope on line 7 column 5  no-shadow
 /* eslint-disable */
 enum SourceType {
-    UNSPLASH = 0,
-    WALLHAVEN,
+    WALLHAVEN = 0,
     REDDIT,
+    UNSPLASH,
     GENERIC_JSON,
     LOCAL_FOLDER,
     STATIC_URL,
```

Comment:
Existing configurations match on these IDs. Scrambling these around will break existing configurations and will require to manually change the type to the correct one.

Quick fresh up on the save behavior:
Each SourceRow has the current timestamp as an ID.
In a 'general' settings group are general settings saved for each SourceRow such as the name, type, blocked images and enabled status.
With those information it's now possible to retrieve the correct adapter details by looking in the general settings for the type and navigating to that type settings group and looking for the ID.


We have a few options:
* Migrate the `sources.general.{ID}.type` schema - either of those:
  * Migrate the enum value
  * Migrate the type to something different like a string to prevent this in the future
* do something like this for backwards compatibility:
    ~~~ diff
    enum SourceType {
        WALLHAVEN = 1,
        REDDIT,
        GENERIC_JSON,
        LOCAL_FOLDER,
        STATIC_URL,
        UNSPLASH,
    }
    ~~~

Reply:
Thanks, good point! I'll look into this but I think moving to string based IDs (maybe the schema id)  is the best choice even when it means braking current configurations. Relying on enum values means that removing sources will break configs unless we fully qualify all values.

Unsplash source itself isn't meant to be backwards compatible because the api is different and not really compatible. I changed the schema ID to not use any previous settings.

Thanks again!